### PR TITLE
DSDEGP-2120 defaults for backfill

### PR DESCRIPTION
**Pull request by tlangs**
### Description

Fixed the error in recovery when data_type isn't defined.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala [hunk 1]
```diff
@@ -73,7 +73,12 @@ class RecoveryIntegrationSpec
       gvcfPath = Some(randomUri(i)),
       documentStatus = Some(DocumentStatus.Normal)
     )
-    gvcfMapper.document(key, metadata)
+    if (i < (documentCount / 2)) {
+      gvcfMapper
+        .document(key, metadata)
+        .mapObject(_.remove("data_type"))
+    } else
+      gvcfMapper.document(key, metadata)
```
Comment (danxmoran) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala]: Can you put brackets around this case, to match the `if` branch?

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala [hunk 2]
```diff
@@ -94,7 +99,12 @@ class RecoveryIntegrationSpec
       cramPath = Some(randomUri(i)),
       documentStatus = Some(DocumentStatus.Normal)
     )
-    cramMapper.document(key, metadata)
+    if (i < (documentCount / 2)) {
+      cramMapper
+        .document(key, metadata)
+        .mapObject(_.remove("data_type"))
+    } else
+      cramMapper.document(key, metadata)
```
Comment (danxmoran) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala]: Same here

File: clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala [hunk 3]
```diff
@@ -67,7 +67,9 @@ class ElasticsearchIndex[+CI <: ClioIndex](
           case s if s.isString =>
             s.asString
           case a if a.isArray || a.isObject =>
-            throw new RuntimeException("Arrays and objects cannot be used as ID fields")
+            throw new RuntimeException(
+              s"Found '$a' for key $key in index $indexName. Arrays and objects cannot be used as ID fields"
```
Comment (danxmoran) [inline clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala]: Can you add the upsert ID to the message, too? That way we can track down the document in storage to see what's up.

File: clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala [hunk 4]
```diff
@@ -136,7 +138,7 @@ object ElasticsearchIndex extends ModelAutoDerivation {
       // Since we compute GCS paths from the ES index name, inconsistency would break GCS paths.
       indexName = "gvcf-v2",
       ElasticsearchFieldMapper.StringsToTextFieldsWithSubKeywords,
-      Json.obj(dataTypeKey -> DataType.WGS.asJson)
+      Json.obj(dataTypeKey -> (DataType.WGS: DataType).asJson)
```
Comment (danxmoran) [inline clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala]: Can you add a comment above this explaining why the type-widening is needed?

**Review by charlesdyan — APPROVED** [review]

**Review by danxmoran — APPROVED** [review]

**Pull request merged**